### PR TITLE
Updating Alluxio and remove deferred initialization of AlluxioMetastoreModule

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
@apc999 Are there any plans to mitigate the CVE-s from https://mvnrepository.com/artifact/org.alluxio/alluxio-shaded-client/2.8.0-2? How serious they are?

**Comment by apc999** [comment]
> @apc999 Are there any plans to mitigate the CVE-s from https://mvnrepository.com/artifact/org.alluxio/alluxio-shaded-client/2.8.0-2? How serious they are?

I am currently on OOO, @beinan  can you help take a look? thanks

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastoreModule.java [hunk 1]
```diff
@@ -47,10 +48,7 @@ protected void setup(Binder binder)
             bindMetastoreModule("thrift", new ThriftMetastoreModule());
             bindMetastoreModule("file", new FileMetastoreModule());
             bindMetastoreModule("glue", new GlueMetastoreModule());
-            // Load Alluxio metastore support through reflection. This makes Alluxio effectively an optional dependency
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastoreModule.java:50]: > Revert "Defer initialization of AlluxioMetastoreModule"

> This only allow to bypass the vulnerability by removing jars. We should
not recommend users to play with jars of plugins. It is very risky
operation. Instead we should fix the Alluxio vulnerabilities.

Being able to tamper with the tarball (remove Alluxio) is still better than nothing.
In my eyes, the Alluxio integration does not look maintained and should perhaps be dropped. Until then, it's probably beneficial to unlock shipping Trino without Alluxio. Of course, such use-case is not recommended, nor supported, but we shouldn't actively work to make it impossible.

**Review by findepi — COMMENTED** [review]
Second commit LGTM

**Comment by beinan** [comment]
> @apc999 Are there any plans to mitigate the CVE-s from https://mvnrepository.com/artifact/org.alluxio/alluxio-shaded-client/2.8.0-2? How serious they are?

Hello @kokosing and @findepi  Both of the CVE-s got fix in alluxio's master, we will release 2.8.1 shortly (probably next week)

https://github.com/Alluxio/alluxio/pull/15666
https://github.com/Alluxio/alluxio/pull/15727

**Comment by kokosing** [comment]
@beinan that is great news. I will wait then for the release. Thank you!

**Comment by beinan** [comment]
> @beinan that is great news. I will wait then for the release. Thank you!

Alluxio 2.8.1 is ready on maven repo, I just upgraded the version, let's see if the build is ok https://github.com/trinodb/trino/pull/13609

**Comment by kokosing** [comment]
updated

**Review by findepi — APPROVED** [review]
"Update alluxio client" LGTM

I'd prefer not to merge the "Revert "Defer initialization of AlluxioMetastoreModule"" commit.
Also, I proposed https://github.com/trinodb/trino/pull/13559 which is related to this.

**Pull request closed without merging**